dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -28,6 +28,8 @@
 
 routes = dashboard_optional_utils.ClassMethodRouteTable
 
+SNAPSHOT_API_TIMEOUT_SECONDS = 30
```

Comment:
Do we want to pass timeout in as a query param for any of these? Similar to the `component_activities` route.

https://github.com/ray-project/ray/blob/de2db06d182aa68a368007db4582087435e3c09a/dashboard/modules/snapshot/snapshot_head.py#L132-L136

That way, if any clusters have long snapshots, we could use a feature flag to target specific clusters to extend the timeout period.